sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -572,6 +572,10 @@ class Dataset[T] private[sql](
     val parsedDelay =
       Option(CalendarInterval.fromString("interval " + delayThreshold))
         .getOrElse(throw new AnalysisException(s"Unable to parse time delay '$delayThreshold'"))
+    // Threshold specified in months/years is non-deterministic
```

Comment:
i worry users think it will align with month boundary. It is safer to just say "31 days".